tests: Add tests for a sequence of updates, and resuming an update

detsch opened the pull request:
@mike-sul This is an ongoing work, not quite ready for review. But it may be useful for you to understand how the changes from #150 are used.

detsch committed:
tests: Add tests for a sequence of updates, and resuming an update

Signed-off-by: Andre Detsch <andre.detsch@foundries.io>

detsch committed:
ci: Add integration tests to CI workflow

Signed-off-by: Andre Detsch <andre.detsch@foundries.io>

mike-sul commented on test/integration/update_test.go:
```diff
@@ -0,0 +1,144 @@
+// that events are posted correctly and status is updated as expected.
+func TestUpdateSequence(t *testing.T) {
+	client.DefaultHttpOperations = mockHttpOperations{}
+	tempDir = t.TempDir()
```
Probably makes sense to pass the `t.TempDir()` to the `mockHttpOperations{}` so we can avoid using the `tempDir` global variable.

detsch committed:
tests: Set custom HttpOperations per GatewayClient instance

Replace usage of global DefaultHttpOperations with instance-specific
HttpOperations in GatewayClient.

Signed-off-by: Andre Detsch <andre.detsch@foundries.io>
